test/SparqlAntlrParserTest.cpp

```diff
@@ -890,62 +890,66 @@ TEST(SparqlParser, ValuesClause) {
-  expectCompleteParse(
-      parseVerbPathOrSimple("test:foo",
-                            {{"test", "<http://www.example.com/>"}}),
-      IsPropertyPath(PropertyPath::fromIri("<http://www.example.com/foo>")));
-  expectCompleteParse(parseVerbPathOrSimple("?bar", {}),
-                      IsPropertyPath(PropertyPath::fromIri("?bar")));
-  expectCompleteParse(
-      parseVerbPathOrSimple(":", {{"", "<http://www.example.com/>"}}),
-      IsPropertyPath(PropertyPath::fromIri("<http://www.example.com/>")));
-  expectCompleteParse(
-      parseVerbPathOrSimple("<http://www.w3.org/1999/02/22-rdf-syntax-ns#type>",
-                            {}),
-      IsPropertyPath(PropertyPath::fromIri(
-          "<http://www.w3.org/1999/02/22-rdf-syntax-ns#type>")));
+  expectVerbPathOrSimple("test:foo",
+                         PropertyPath::fromIri("<http://www.example.com/foo>"),
+                         {{"test", "<http://www.example.com/>"}});
+  expectVerbPathOrSimple("?bar", Variable{"?bar"});
+  expectVerbPathOrSimple(":",
+                         PropertyPath::fromIri("<http://www.example.com/>"),
+                         {{"", "<http://www.example.com/>"}});
+  expectVerbPathOrSimple(
+      "<http://www.w3.org/1999/02/22-rdf-syntax-ns#type>",
+      PropertyPath::fromIri(
+          "<http://www.w3.org/1999/02/22-rdf-syntax-ns#type>"));
   // Test the basic combinators / | (...) + * ?.
-  expectCompleteParse(
-      parseVerbPathOrSimple("a:a / a:b", {{"a", "<http://www.example.com/>"}}),
-      IsPropertyPath(PropertyPath::makeSequence(
+  expectVerbPathOrSimple(
+      "a:a / a:b",
+      PropertyPath::makeSequence(
           {PropertyPath::fromIri("<http://www.example.com/a>"),
-           PropertyPath::fromIri("<http://www.example.com/b>")})));
-  expectCompleteParse(
-      parseVerbPathOrSimple("a:a | a:b", {{"a", "<http://www.example.com/>"}}),
-      IsPropertyPath(PropertyPath::makeAlternative(
+           PropertyPath::fromIri("<http://www.example.com/b>")}),
+      {{"a", "<http://www.example.com/>"}});
+  expectVerbPathOrSimple(
+      "a:a | a:b",
+      PropertyPath::makeAlternative(
           {PropertyPath::fromIri("<http://www.example.com/a>"),
-           PropertyPath::fromIri("<http://www.example.com/b>")})));
-  expectCompleteParse(
-      parseVerbPathOrSimple("(a:a)", {{"a", "<http://www.example.com/>"}}),
-      IsPropertyPath(
-          PropertyPath(PropertyPath::fromIri("<http://www.example.com/a>"))));
-  expectCompleteParse(
-      parseVerbPathOrSimple("a:a+", {{"a", "<http://www.example.com/>"}}),
-      IsPropertyPath(PropertyPath::makeTransitiveMin(
-          {PropertyPath::fromIri("<http://www.example.com/a>")}, 1)));
+           PropertyPath::fromIri("<http://www.example.com/b>")}),
+      {{"a", "<http://www.example.com/>"}});
+  expectVerbPathOrSimple(
+      "(a:a)",
+      PropertyPath(PropertyPath::fromIri("<http://www.example.com/a>")),
```

Comment:
Isn't the outer `PropertyPath` redundant?
A good example for stuff we see when we have clearer abstractions.

Reply:
Yes.